### PR TITLE
Render text questions

### LINKED ISSUE #6916: [North Star Applicant UI] Render text questions
#### Describe the solution presented in this issue.
Render text questions in thymeleaf using USWDS input

#### Done when:
What criteria need to be met for the issue to be finished? 
- [ ] Renders text questions, including validation errors, capturing values when submitted

### PR DESCRIPTION
**Pull request by greg-hyde**
### Description

Thymeleaf rendering of text questions

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)

- [x] Added the correct label: < feature | enhancement | bug | under-development | dependencies | infrastructure | ignore-for-release | database >
- [x] Assigned to a specific person, `civiform/developers`, or a [more specific round-robin list](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-reviewers)
- [x] Created unit and/or browser tests which fail without the change (if possible)
- [x] Performed manual testing (Chrome and Firefox if it includes front-end changes)

#### User visible changes

  - [x] Didn't use a message in applicant facing code that isn't translated yet (unless behind a flag)
- [x] Tested on mobile view. See [mobile device mode](https://developer.chrome.com/docs/devtools/device-mode/)
- [x] Manually tested at 200% size
- [x] Manually evaluated tab order

#### New Features

- [x] Conditioned new functionality on a [FeatureFlag](https://github.com/civiform/civiform/wiki/Feature-Flags)
- [x] Wrote browser tests with the feature flag off and on, etc.

### Issue(s) this completes

Fixes #6916

### CODE REVIEW

**Comment by greg-hyde** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/civiform/civiform/6900?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#6900** <a href="https://app.graphite.dev/github/pr/civiform/civiform/6900?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#6715** <a href="https://app.graphite.dev/github/pr/civiform/civiform/6715?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#6710** <a href="https://app.graphite.dev/github/pr/civiform/civiform/6710?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#6702** <a href="https://app.graphite.dev/github/pr/civiform/civiform/6702?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 2 other dependent PRs ([#6703](https://github.com/civiform/civiform/pull/6703) <a href="https://app.graphite.dev/github/pr/civiform/civiform/6703?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#6867](https://github.com/civiform/civiform/pull/6867) <a href="https://app.graphite.dev/github/pr/civiform/civiform/6867?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#6701** <a href="https://app.graphite.dev/github/pr/civiform/civiform/6701?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#6881](https://github.com/civiform/civiform/pull/6881) <a href="https://app.graphite.dev/github/pr/civiform/civiform/6881?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#6700** <a href="https://app.graphite.dev/github/pr/civiform/civiform/6700?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#6513** <a href="https://app.graphite.dev/github/pr/civiform/civiform/6513?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 4 other dependent PRs ([#6514](https://github.com/civiform/civiform/pull/6514) <a href="https://app.graphite.dev/github/pr/civiform/civiform/6514?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#6531](https://github.com/civiform/civiform/pull/6531) <a href="https://app.graphite.dev/github/pr/civiform/civiform/6531?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#6581](https://github.com/civiform/civiform/pull/6581) <a href="https://app.graphite.dev/github/pr/civiform/civiform/6581?utm_source=stack-comment-icon" target="_blank">Graphite</a> and 1 other)
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @greg-hyde and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: server/app/views/questiontypes/TextQuestionFragment.html [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    <div th:replace="~{questiontypes/QuestionBaseFragment :: validationErrors(${textQuestion.getValidationErrors().get(question.getContextualizedPath())})}"></div>
+    
+    <input type="text"
+        class="usa-input"
+        th:aria-describedby="${questionId} + '-description'"
+        maxlength="10000"
+        th:name="${textQuestion.getTextPath()}"
+        th:value="${textQuestion.getTextValue().orElse('')}"
+        th:aria-required="${!question.isOptional()}"
+        th:aria-invalid="${!textQuestion.getValidationErrors().isEmpty()}">
```
Comment (gwendolyngoetz) [inline server/app/views/questiontypes/TextQuestionFragment.html:16]: suggestion (non-blocking): group the aria attributes together

File: server/app/views/questiontypes/TextQuestionFragment.html [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<div th:fragment="text (question)"
+     th:with="textQuestion=${question.createTextQuestion()}, questionId=${#strings.randomAlphanumeric(8)}">
+    <!-- Title and Help Text -->
+    <div th:replace="~{questiontypes/QuestionBaseFragment :: titleAndHelpText(${question}, ${questionId})}"></div>
```
Comment (gwendolyngoetz) [inline server/app/views/questiontypes/TextQuestionFragment.html:4]: question: Should this be a label or should there be a label somewhere?

File: browser-test/src/applicant/questions/text.test.ts [hunk 3]
```diff
@@ -31,6 +33,11 @@ test.describe('Text question for applicant flow', () => {
       await logout(page)
     })
 
+    test.beforeEach(async () => {
+      const {page} = ctx
+      await disableFeatureFlag(page, 'north_star_applicant_ui')
+    })
```
Comment (gwendolyngoetz) [inline browser-test/src/applicant/questions/text.test.ts:39]: Unless you have the flag set to be enabled in `/server/conf/application.dev-browser-tests.conf` you shouldn't need this. The browser context is refresh in a beforeeach (index.tx) and the databases is dropped in an aftereach.

**Review by gwendolyngoetz — COMMENTED** [review]
Are you able to include a screenshot that shows the validation message too?

**Comment by greg-hyde** [comment]
> Are you able to include a screenshot that shows the validation message too?

good catch I wasn't able to do that earlier because the submit button didn't exist yet :) done

File: server/app/views/questiontypes/TextQuestionFragment.html (hunk 2 above)
Comment (greg-hyde) [reply server/app/views/questiontypes/TextQuestionFragment.html:4]: This div gets replaced by the contents of that template. Right now that template uses a <legend> but I think I need to do a small refactor to make it conditionally a label or legend depending on whether it has one or multiple inputs. I was thinking of making that its own PR, but I could do it here if you prefer.

File: server/app/views/questiontypes/TextQuestionFragment.html (hunk 1 above)
Comment (greg-hyde) [reply server/app/views/questiontypes/TextQuestionFragment.html:16]: done

File: server/app/views/questiontypes/TextQuestionFragment.html (hunk 2 above)
Comment (gwendolyngoetz) [reply server/app/views/questiontypes/TextQuestionFragment.html:4]: Separate is fine

File: server/app/views/questiontypes/TextQuestionFragment.html (hunk 1 above)
Comment (gwendolyngoetz) [reply server/app/views/questiontypes/TextQuestionFragment.html:16]: Doesn't look like it took. `aria-describedby` is still up

File: browser-test/src/applicant/questions/text.test.ts (hunk 3 above)
Comment (greg-hyde) [reply browser-test/src/applicant/questions/text.test.ts:39]: if I remove this, I'm seeing the tests after the north star tests fail...

Comment (gwendolyngoetz) [reply browser-test/src/applicant/questions/text.test.ts:39]: Hmm. Ok. Maybe that index code isn't clearing something else after all. Appreciate you trying. Thanks

**Review by gwendolyngoetz — APPROVED** [review]

**Pull request merged**